dnf5/main.cpp

```diff
@@ -784,18 +786,6 @@ int main(int argc, char * argv[]) try {
         }
         std::cerr << ex.what() << std::endl;
         return static_cast<int>(libdnf::cli::ExitCode::ERROR);
-    } catch (libdnf::cli::ArgumentParserMissingCommandError & ex) {
-        // print help if no command is provided
-        std::cerr << ex.what() << std::endl;
-        context.get_argument_parser().get_selected_command()->help();
-        return static_cast<int>(libdnf::cli::ExitCode::ARGPARSER_ERROR);
-    } catch (libdnf::cli::ArgumentParserMissingDependentArgumentError & ex) {
-        std::cerr << ex.what() << std::endl;
-        context.get_argument_parser().get_selected_command()->help();
-        return static_cast<int>(libdnf::cli::ExitCode::ARGPARSER_ERROR);
-    } catch (libdnf::cli::ArgumentParserError & ex) {
-        std::cerr << ex.what() << std::endl;
-        return static_cast<int>(libdnf::cli::ExitCode::ARGPARSER_ERROR);
```

Comment:
Removing the `catch` statements evokes that ArgumentParserError may actually never be raised in the corresponding `try` statement.  Does it mean that we should never ever `throw` it in `command->run()` etc.?

Reply:
@praiskup  I thought about it. There is at least one case where we throw an exception in command->configure (a problem with the download command). So I will modify the PR.

Reply:
@m-blaha Good point. Thanks.